File: mediscreen-front/src/app/patient/model/Patient.ts
export interface PatientElement {
  idPatient: any;
  lastName: string;
  firstName: string;
  dateOfBirth: string;
  address: string;
  phoneNumber: number;
  sex: string;
}

export class Patient implements PatientElement {

  idPatient: any;
  lastName: string;
  firstName: string;
  dateOfBirth: string;
  address: string;
  phoneNumber: number;
  sex: string;

  constructor(idPatient: any, lastName: string, firstName: string, dateOfBirth: string, address: string, phoneNumber: number, sex: string) {
    this.idPatient = idPatient;
    this.lastName = lastName;
    this.firstName = firstName;
    this.dateOfBirth = dateOfBirth;
    this.address = address;
    this.phoneNumber = phoneNumber;
    this.sex = sex;
  }

}

File: patient/src/main/java/com/openclassrooms/patient/entity/Patient.java
package com.openclassrooms.patient.entity;

import com.sun.istack.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "patient")
public class Patient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idPatient;


    @Column(name = "last_name")
    @NotNull
    private String lastName;

    @Column(name = "first_name")
    @NotNull
    private String firstName;

    @Column(name = "date_birth")
    @DateTimeFormat(pattern = "YYYY-mm-dd")
    private LocalDate dateOfBirth;

    @Column(name = "address")
    private String address;

    @Column(name = "phone_number")
    private long phoneNumber;

    @Column(name = "sex")
    private String sex;

}
